client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -271,6 +271,29 @@ function PatternPreviewFragment( {
 		};
 	} );
 
+	// This fetches forms and adds a listener that disables submission
+	useEffect( () => {
+		const iframe = ref.current?.querySelector( 'iframe' );
+
+		if ( ! iframe?.contentDocument ) {
+			return;
+		}
+
+		const forms = Array.from( iframe.contentDocument.querySelectorAll( 'form' ) );
+
+		if ( ! forms.length ) {
+			return;
+		}
+
+		const onFormSubmit = ( event: { preventDefault: () => void } ) => event.preventDefault();
```

Comment:
```suggestion
		const onFormSubmit = ( event: SubmitEvent ) => event.preventDefault();
```
`SubmitEvent` is the type TS infers for `submit` events.